lib/google_crawler_web/controllers/keyword_controller.ex

```diff
@@ -1,12 +1,17 @@
 defmodule GoogleCrawlerWeb.KeywordController do
   use GoogleCrawlerWeb, :controller
 
-  alias GoogleCrawler.{Repo, Search}
-  alias GoogleCrawler.Search.Keyword
+  alias GoogleCrawler.Repo
+  alias GoogleCrawler.Search.{Keyword, KeywordQuery, KeywordSearchForm}
 
-  def index(conn, _params) do
-    keywords = Search.list_keywords(conn.assigns[:current_user])
-    render(conn, "index.html", keywords: keywords)
+  def index(conn, params) do
+    search_form_changeset =
+      KeywordSearchForm.changeset(%KeywordSearchForm{}, params["filter"] || %{})
+
+    search_form_params = Ecto.Changeset.apply_changes(search_form_changeset)
```

Comment:
How about moving this logic to `GoogleCrawler.Search.KeywordSearchForm` to make controller cleaner?
```elixir
# lib/google_crawler/search/keyword_search_form.ex
def new(params) do
  %__MODULE__{}
  |> changeset(params)
  |> Ecto.Changeset.apply_changes()
end

defp changeset() do
  # ...
end

# lib/google_crawler_web/controllers/keyword_controller.ex
def index() do
  search_form = KeywordSearchForm.new(params)
end
```

Reply:
Nice one, thanks for the suggestion. I extracted it out of controller